src/machine/machine_rp2040_rng.go

```diff
@@ -0,0 +1,36 @@
+//go:build rp2040
+// +build rp2040
+
+// Implementation based on code located here:
+// https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_lwip/random.c
+
+package machine
+
+import (
+	"device/rp"
+)
+
+const numberOfCycles = 32
+
+// GetRNG returns 32 bits of semi-random data based on ring oscillator.
+func GetRNG() (uint32, error) {
+	var val uint32
+	for i := 0; i < 4; i++ {
+		val = (val << 8) | uint32(roscRandByte())
+	}
+	return val, nil
+}
+
+func roscRandByte() uint8 {
+	var val, poly uint8
```

Comment:
The reference code has `val` being a static byte that's persistent across calls.  Did you want to make this a global to achieve the same effect?

Reply:
Changed, because that is in fact what I want.

Thanks for pointing that out @dgryski